ocaml/tests/storage_impl_test.ml

```diff
@@ -1,535 +0,0 @@
-  Debug_print_impl.VDI.working := true;
-  (* Should succeed because the VDI.attach state will have been forgotten: FH2 *)
-  expect "()" (fun x -> x = ())
-    (Client.SR.detach ~dbg ~sr)
-
-let test_sr_detach_cleanup_errors_2 sr vdi =
-  Debug_print_impl.VDI.working := false;
-  expect "()" (fun x -> x = ())
-    (Client.SR.attach ~dbg ~sr ~device_config:[]);
-  let dp = datapath_of_id "datapath" in
-  leak dp sr true vdi;
-  if vdi = "error2"
-  then expect "backend_error" backend_error
-      (fun () -> Client.DP.destroy ~dbg ~dp ~allow_leak:false)
-  else expect "internal_error" internal_error
-      (fun () -> Client.DP.destroy ~dbg ~dp ~allow_leak:false);
-  Debug_print_impl.VDI.working := true;
-  debug "Attempting to attach RO (having failed a detach of a RW detach)";
-  expect "Params _" (function _ -> true)
-    (Client.VDI.attach ~dbg ~dp ~sr ~vdi ~read_write:false);
-  debug "Detaching and cleaning up";
-  expect "()" (fun x -> x = ())
-    (Client.SR.detach ~dbg ~sr)
-
-let create_vdi_test sr =
-  let dp = datapath_of_id "datapath" in
-  expect "()" (fun x -> x = ())
-    (Client.SR.attach ~dbg ~sr ~device_config:[]);
-  let vdi_info = {
-    vdi = "";
-    uuid = None;
-    content_id = "";
-    name_label = "name_label";
-    name_description = "name_description";
-    virtual_size = 10L;
-    ty = "user";
-    is_a_snapshot = false;
-    snapshot_time = "";
-    snapshot_of = "";
-    read_only = false;
-    cbt_enabled = false;
-    physical_utilisation = 10L;
-    metadata_of_pool = "";
-    persistent = true;
-    sm_config = [];
-  } in
-  expect "too_small_backend_error" too_small_backend_error
-    (fun () ->
-       let vdi_info = { vdi_info with sm_config = ["toosmall", ""] } in
-       Client.VDI.create ~dbg ~sr ~vdi_info);
-  let vdi = Client.VDI.create ~dbg ~sr ~vdi_info in
-  expect "attach_info" (fun _ -> true)
-    (Client.VDI.attach ~dbg ~dp ~sr ~vdi:vdi.vdi ~read_write:false);
-  debug "Detaching and cleaning up";
-  expect "()" (fun x -> x = ())
-    (Client.SR.detach ~dbg ~sr)
-
-let _ =
-  Storage_impl.print_debug := true;
-  Storage_impl.host_state_path := "/tmp/storage.db";
-  Vdi_automaton.test ();
-  Unixext.unlink_safe !Storage_impl.host_state_path;
-  let s = Xcp_service.make ~path:Xapi_globs.storage_unix_domain_socket ~queue_name:"org.xen.xapi.storage" ~rpc_fn:(Server.process None) () in
-  info "Started service on org.xen.xapi.storage";
-  let (_: Thread.t) = Thread.create (fun () -> Xcp_service.serve_forever s) () in
-
-  info "Listening on %s" Xapi_globs.storage_unix_domain_socket;
-  test_sr "sr";
-
-  test_sr_detach_cleanup_errors_1 "sr" "error2";
-  test_sr_detach_cleanup_errors_1 "sr" "error";
-
-  test_sr_detach_cleanup_errors_2 "sr" "error2";
-  test_sr_detach_cleanup_errors_2 "sr" "error";
-
-  test_stat "sr" "vdi";
-
-  create_vdi_test "sr";
-
-  if !total_errors = 0 then begin
-    info "OK";
-    exit 0;
-  end else begin
-    info "%d errors detected" !total_errors;
-    exit 1
-  end
-
```

Comment:
This test looks useful, do we have equivalents for it in quicktest, or can we add this to quicktest?

Reply:
No, but it's a layering violation. I think it's not worth the effort. But we can still add a ticket to re-add it.

Reply:
Ok lets drop this test.